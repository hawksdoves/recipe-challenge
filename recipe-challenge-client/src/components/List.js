import React from 'react';
import { Link } from 'react-router-dom';
import { Card } from 'semantic-ui-react';

import Recipe from './Recipe';
import Favourite from './Favourite';

function List(props) {
    const { recipes, favourites, ...rest } = props;

    return recipes.length > 0 ?
        (<Card.Group >
            { recipes.map(recipe => {
                return (  
                    <div key={recipe.id} className='recipe' >
                        <Link to={`/recipe/${recipe.id}`} key={`Link-${recipe.id}`} className='card'>
                            <Recipe recipe={recipe} />
                        </Link>
                        <Favourite favourite={favourites.includes(recipe.id)} {...rest} recipeId={recipe.id} />
                    </div>
                )
            })}
        </Card.Group> ) :
        <div>Sorry, nothing matched your filter term</div>
};

export default List;
