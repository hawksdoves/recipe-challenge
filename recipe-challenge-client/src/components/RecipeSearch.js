import React from 'react';
import { Icon, Input } from 'semantic-ui-react';

function RecipeSearch(props) {
    const { searchRecipes } = props;

    return <Input 
                className='recipeSearch' 
                type="text" 
                placeholder="Recipe Search..." 
                onChange={searchRecipes} 
                icon={<Icon name='search' inverted circular link />} 
            />
}

export default RecipeSearch;
