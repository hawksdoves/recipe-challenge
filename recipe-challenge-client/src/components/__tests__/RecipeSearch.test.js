import React from 'react';
import { shallow } from 'enzyme';

import RecipeSearch from '../RecipeSearch';
import List from '../List';

describe('RecipeSearch', () => {
    const recipes = [
        {
            id: '1',
            name: 'steak',
            ingredients: [{
                name: 'steak',
                quantity: '3'
            }]
        },
        {
            id: '2',
            name: 'chicken',
            ingredients: [{
                name: 'chicken',
                quantity: '2'
            }]
        }
    ];

    it('renders children', () => {
        const wrapper = shallow(<RecipeSearch recipes={recipes} children={<List/>}/>);

        expect(wrapper).toMatchSnapshot();
    })

    it('should call searchRecipes on change', () => {
        const mockSearchRecipes = jest.fn();
        const wrapper = shallow(<RecipeSearch recipes={recipes} searchRecipes={mockSearchRecipes} />);
        wrapper.find('.recipeSearch').simulate('change', { target: { value: 'steak' } });

        expect(mockSearchRecipes).toHaveBeenCalled();
    })

    it('renders error message if search had no matches', () => {
        const wrapper = shallow(<RecipeSearch recipes={recipes}/>);
        wrapper.find('.recipeSearch').simulate('change', { target: { value: 'fish' } });
        
        expect(wrapper.find(List).length).toEqual(0);
        expect(wrapper).toMatchSnapshot();
    })
});
