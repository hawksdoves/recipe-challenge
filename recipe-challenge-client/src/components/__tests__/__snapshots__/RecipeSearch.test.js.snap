// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeSearch renders children 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Input
    className="recipeSearch"
    icon={
        <Icon
            as="i"
            circular={true}
            inverted={true}
            link={true}
            name="search"
        />
    }
    onChange={undefined}
    placeholder="Recipe Search..."
    type="text"
/>,
  "nodes": Array [
    <Input
      className="recipeSearch"
      icon={
            <Icon
                  as="i"
                  circular={true}
                  inverted={true}
                  link={true}
                  name="search"
            />
      }
      onChange={undefined}
      placeholder="Recipe Search..."
      type="text"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <RecipeSearch
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                          "name": "chicken",
                        },
                      ]
        }
>
        <List />
</RecipeSearch>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <List />,
          "recipes": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Input
          className="recipeSearch"
          icon={
                    <Icon
                              as="i"
                              circular={true}
                              inverted={true}
                              link={true}
                              name="search"
                    />
          }
          onChange={undefined}
          placeholder="Recipe Search..."
          type="text"
/>,
        "_debugID": 2,
        "_renderedOutput": <Input
          className="recipeSearch"
          icon={
                    <Icon
                              as="i"
                              circular={true}
                              inverted={true}
                              link={true}
                              name="search"
                    />
          }
          onChange={undefined}
          placeholder="Recipe Search..."
          type="text"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecipeSearch
    recipes={
        Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ]
    }
>
    <List />
</RecipeSearch>,
}
`;

exports[`RecipeSearch renders error message if search had no matches 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Input
    className="recipeSearch"
    icon={
        <Icon
            as="i"
            circular={true}
            inverted={true}
            link={true}
            name="search"
        />
    }
    onChange={undefined}
    placeholder="Recipe Search..."
    type="text"
/>,
  "nodes": Array [
    <Input
      className="recipeSearch"
      icon={
            <Icon
                  as="i"
                  circular={true}
                  inverted={true}
                  link={true}
                  name="search"
            />
      }
      onChange={undefined}
      placeholder="Recipe Search..."
      type="text"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <RecipeSearch
        recipes={
                Array [
                        Object {
                          "id": "1",
                          "ingredients": Array [
                            Object {
                              "name": "steak",
                              "quantity": "3",
                            },
                          ],
                          "name": "steak",
                        },
                        Object {
                          "id": "2",
                          "ingredients": Array [
                            Object {
                              "name": "chicken",
                              "quantity": "2",
                            },
                          ],
                          "name": "chicken",
                        },
                      ]
        }
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "recipes": Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Input
          className="recipeSearch"
          icon={
                    <Icon
                              as="i"
                              circular={true}
                              inverted={true}
                              link={true}
                              name="search"
                    />
          }
          onChange={undefined}
          placeholder="Recipe Search..."
          type="text"
/>,
        "_debugID": 6,
        "_renderedOutput": <Input
          className="recipeSearch"
          icon={
                    <Icon
                              as="i"
                              circular={true}
                              inverted={true}
                              link={true}
                              name="search"
                    />
          }
          onChange={undefined}
          placeholder="Recipe Search..."
          type="text"
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RecipeSearch
    recipes={
        Array [
            Object {
              "id": "1",
              "ingredients": Array [
                Object {
                  "name": "steak",
                  "quantity": "3",
                },
              ],
              "name": "steak",
            },
            Object {
              "id": "2",
              "ingredients": Array [
                Object {
                  "name": "chicken",
                  "quantity": "2",
                },
              ],
              "name": "chicken",
            },
          ]
    }
/>,
}
`;
